### PR TITLE
8278325: excluded class should not be checked again for exclusion

### PR DESCRIPTION
**Pull request by yminqi**
Hi, Please review

  When LambdaFormInvoker regenerate lambda form holder class, the old class with the same name already loaded and is marked for excluded for dump. This class should not be checked against exclusion, or it will output unexpected warning like:
[0.394s][warning][cds] Skipping java/lang/invoke/BoundMethodHandle$Species_J: Unsupported location
[0.394s][warning][cds] Skipping java/lang/invoke/BoundMethodHandle$Species_JL: Unsupported location
[0.394s][warning][cds] Skipping java/lang/invoke/BoundMethodHandle$Species_FL: Unsupported location
[0.394s][warning][cds] Skipping java/lang/invoke/BoundMethodHandle$Species_F: Unsupported location
  The fix changed the order for checking exclusion of a class --- only check for those that have not been set for exclusion. Original function DumpTimeClassInfo::is_excluded in fact is checking three conditions in logical OR, it does not tell which reason for exclusion. In this fix, we need check the exact reason which is set for _excluded. Original is_excluded is renamed to should_be_excluded.

Tests: tier1,tier4 (in testing)

Thanks
Yumin

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8278325](https://bugs.openjdk.java.net/browse/JDK-8278325): excluded class should not be checked again for exclusion

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7225/head:pull/7225` \
`$ git checkout pull/7225`

Update a local copy of the PR: \
`$ git checkout pull/7225` \
`$ git pull https://git.openjdk.java.net/jdk pull/7225/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7225`

View PR using the GUI difftool: \
`$ git pr show -t 7225`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7225.diff">https://git.openjdk.java.net/jdk/pull/7225.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/classfile/systemDictionaryShared.cpp [hunk 1]
```diff
@@ -1230,7 +1232,7 @@ class CopyLambdaProxyClassInfoToArchive : StackObj {
     // The following check has been moved to SystemDictionaryShared::check_excluded_classes(), which
     // happens before the classes are copied.
     //
-    // if (SystemDictionaryShared::is_excluded_class(info._proxy_klasses->at(0))) {
+    // if (SystemDictionaryShared::should_be_excluded_class(info._proxy_klasses->at(0))) {
```
Comment (iklam) [inline src/hotspot/share/classfile/systemDictionaryShared.cpp:1235]: The change seems unrelated to the PR. We don't have a function named `SystemDictionaryShared::should_be_excluded_class`

File: test/hotspot/jtreg/runtime/cds/appcds/TestDumpClassListSource.java [hunk 2]
```diff
@@ -117,9 +118,28 @@ public static void main(String[] args) throws Exception {
         checkFileExistence("Archive", fileArchive);
         checkFileExistence("ClassList", fileList);
 
-        output.shouldHaveExitValue(0);
+        output.shouldHaveExitValue(0)
+              .shouldContain(warningMessage);
```
Comment (iklam) [inline test/hotspot/jtreg/runtime/cds/appcds/TestDumpClassListSource.java:122]: Why would the above process generate the "Skipping java/lang/invoke/BoundMethodHandle$Species_" message?

**Review by iklam — COMMENTED** [review]
Hi Yumin, due to the history of the bug report (I thought the bug was in a separate place but there was a problem in my test case), it's very difficult to understand what the actual bug is.

Could you close JDK-8278325 as "Not an issue", and file a new bug for the problem that you're trying to fix (preferably with a simple test case that demonstrates the problem)?

Thanks

File: src/hotspot/share/classfile/systemDictionaryShared.cpp (hunk 1 above)
Comment (yminqi) [reply src/hotspot/share/classfile/systemDictionaryShared.cpp:1235]: This is a mistake -- should not be changed.

File: test/hotspot/jtreg/runtime/cds/appcds/TestDumpClassListSource.java (hunk 2 above)
Comment (yminqi) [reply test/hotspot/jtreg/runtime/cds/appcds/TestDumpClassListSource.java:122]: This is dynamic dump, the problem is filed against static dump. When do dynamic dump we did not have a those in SPECIES_RESOLVE in classlist like static dump. They are not regenerated in fact.
